client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/IndicesClientDocumentationIT.java

```diff
@@ -545,5 +548,126 @@ public void onFailure(Exception e) {
             });
             // end::update-aliases-execute-async
         }
-    }    
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testShrinkIndex() throws IOException {
+        RestHighLevelClient client = highLevelClient();
+
+        {
+            Map<String, Object> nodes = getAsMap("_nodes");
+            String firstNode = ((Map<String, Object>) nodes.get("nodes")).keySet().iterator().next();
+            createIndex("source_index", Settings.builder().put("index.number_of_shards", 4).put("index.number_of_replicas", 0).build());
+            updateIndexSettings("source_index", Settings.builder().put("index.routing.allocation.require._name", firstNode)
+                    .put("index.blocks.write", true));
+        }
+
+        // tag::shrink-index-request
+        ResizeRequest request = new ResizeRequest("target_index","source_index"); // <1>
+        // end::shrink-index-request
+
+        // tag::shrink-index-request-timeout
+        request.timeout(TimeValue.timeValueMinutes(2)); // <1>
+        request.timeout("2m"); // <2>
+        // end::shrink-index-request-timeout
+        // tag::shrink-index-request-masterTimeout
+        request.masterNodeTimeout(TimeValue.timeValueMinutes(1)); // <1>
+        request.masterNodeTimeout("1m"); // <2>
+        // end::shrink-index-request-masterTimeout
+        // tag::shrink-index-request-waitForActiveShards
+        request.getTargetIndexRequest().waitForActiveShards(2); // <1>
+        request.getTargetIndexRequest().waitForActiveShards(ActiveShardCount.DEFAULT); // <2>
+        // end::shrink-index-request-waitForActiveShards
+        // tag::shrink-index-request-settings
+        request.getTargetIndexRequest().settings(Settings.builder().put("index.number_of_shards", 2)); // <1>
+        // end::shrink-index-request-settings
+        // tag::shrink-index-request-aliases
+        request.getTargetIndexRequest().alias(new Alias("target_alias")); // <1>
+        // end::shrink-index-request-aliases
+
+        // tag::shrink-index-execute
+        ResizeResponse resizeResponse = client.indices().shrink(request);
+        // end::shrink-index-execute
+
+        // tag::shrink-index-response
+        boolean acknowledged = resizeResponse.isAcknowledged(); // <1>
+        boolean shardsAcked = resizeResponse.isShardsAcknowledged(); // <2>
+        // end::shrink-index-response
+        assertTrue(acknowledged);
+        assertTrue(shardsAcked);
+
+        // tag::shrink-index-execute-async
+        client.indices().shrinkAsync(request, new ActionListener<ResizeResponse>() {
+            @Override
+            public void onResponse(ResizeResponse resizeResponse) {
+                // <1>
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                // <2>
+            }
+        });
+        // end::shrink-index-execute-async
+    }
```

Comment:
I think that this is OK here because the test framework will wait for pending cluster state updates to be processed before executing the next test and resize action is based on cluster state updates.